samples/dnn/object_detection.py

```diff
@@ -163,9 +160,23 @@ def drawPred(classId, conf, left, top, right, bottom):
         print('Unknown output layer type: ' + lastLayer.type)
         exit()
 
-    indices = cv.dnn.NMSBoxes(boxes, confidences, confThreshold, nmsThreshold)
+    indices = []
+    if lastLayer.type == 'Region' and len(outNames) > 1:
+        classIds = np.array(classIds)
+        boxes = np.array(boxes)
+        confidences = np.array(confidences)
+        unique_classes = set(classIds)
+        for cl in unique_classes:
+            class_indices = np.where(classIds == cl)[0]
+            conf = confidences[class_indices]
+            box  = boxes[class_indices].tolist()
+            nms_indices = cv.dnn.NMSBoxes(box, conf, confThreshold, nmsThreshold)
```

Comment:
Can you please check if we really need NMS for YOLOv3? I mean if NMS in 3 Region layers is not enough.

Reply:
Added proof to description